config/configmapprovider/provider.go

```diff
@@ -66,6 +68,15 @@ type Provider interface {
 	Shutdown(ctx context.Context) error
 }
 
+// CloseFunc signals that the configuration for which it was used to retrieve values is
+// no longer in use and should close and release any watchers that it may have created.
+//
+// Should block until all resources are closed, and guarantee that `onChange` is not
+// going to be called after it returns except when `ctx` is cancelled.
+//
+// TODO: Find a good name for this func.
```

Comment:
Any ideas?

Reply:
This feels like a reasonable name.  Externally, it'll be referenced as `configmapprovider.CloseFunc`, which is a bit of a mouthful but has all the relevant info.

If this doesn't feel right, what about a `Close` method on the `Provider` interface?

Reply:
I tried `StopWatchFunc`, if not better I can revert.

Reply:
That makes me think of a timer/stopwatch.  Maybe `EndWatchFunc`?

Reply:
Happy to try that, between `EndWatchFunc` or `CloseWatchFunc` or `DoneFunc` (does not seem to end, but saw this pattern)?

Reply:
`DoneFunc` also seems good.  I think I would be happy with either that or `EndWatchFunc`.  Would like to know what others think.

Reply:
FYI: Usually (for unknown reasons) you don't get feedback unless you tag people :) @tigrannajaryan @codeboten @jpkrohling @dmitryax

Reply:
Tried `EndWatchFunc` looks ok...